code/_onclick/hud/screen_objects.dm

```diff
@@ -520,35 +524,39 @@
 	if(mods["alt"])
 		var/list/options = list()
 		if(user.hive.living_xeno_queen)
-			options["Queen"] = list(TRACKER_QUEEN, 0)
+			// Don't need weakrefs to this or the hive core, since there's only one possible target.
+			options["Queen"] = list(null, TRACKER_QUEEN)
 
 		if(user.hive.hive_location)
-			options["Hive Core"] = list(TRACKER_HIVE, 0)
+			options["Hive Core"] = list(null, TRACKER_HIVE)
 
-		var/xeno_leader_index = 1
-		for(var/xeno in user.hive.xeno_leader_list)
-			var/mob/living/carbon/xenomorph/xeno_lead = user.hive.xeno_leader_list[xeno_leader_index]
-			if(xeno_lead)
-				options["Xeno Leader [xeno_lead]"] = list(TRACKER_LEADER, xeno_leader_index)
-			xeno_leader_index++
+		for(var/mob/living/carbon/xenomorph/leader in user.hive.xeno_leader_list)
+			options["Xeno Leader [leader]"] = list(WEAKREF(leader), TRACKER_LEADER)
 
 		var/list/sorted_tunnels = sort_list_dist(user.hive.tunnels, get_turf(user))
-		var/tunnel_index = 1
-		for(var/obj/structure/tunnel/tunnel in sorted_tunnels)
-			options["Tunnel [tunnel.tunnel_desc]"] = list(TRACKER_TUNNEL, tunnel_index)
-			tunnel_index++
+		for(var/obj/structure/tunnel/tunnel as anything in sorted_tunnels)
+			options["Tunnel [tunnel.tunnel_desc]"] = list(WEAKREF(tunnel), TRACKER_TUNNEL)
```

Comment:
Lets only convert them to weakrefs when you've actually picked one.

It can be a lot of extra work that we don't really need, and the reference should only last as long as the input list is open (feel free to add like a 30 SECONDS or whatever timeout to the tgui_input_list call if you want).

If you're concerned about making a weakref to something that got deleted between the time the person was making a selection, you can just ignore that (it'll make a weakref that on first resolve then your weakref will return null on for you to untrack) or you can add an extra if QDELETED check when we're making a selection.

Reply:
Yeah that would work better, thanks for the suggestion!

I don't think it'll be necessary to add any extra checks to it, since like you said if the atom gets deleted it'll try to resolve, fail, then revert back to tracking the queen.